cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -752,83 +754,92 @@ std::vector<row_range> compute_page_splits_by_row(device_span<cumulative_page_in
   return splits;
 }
 
+struct codec_stats {
+  Compression compression_type  = UNCOMPRESSED;
+  size_t num_pages              = 0;
+  int32_t max_decompressed_size = 0;
+  size_t total_decomp_size      = 0;
+
+  void add_pages(host_span<ColumnChunkDesc const> chunks,
```

Comment:
Looks like this function would pick dict or data pages from the input pages based on `select_dict_pages`. Maybe `select_dict_pages` should be a strong type or the function may have a more descriptive name.